### ISSUE TITLE
Disallow environment variable in CMake

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#365]
I mean the QMCPACK specific ones not the libraries.

I saw the commend in CMakeLists.txt
\# todo: disallow environment
QMC_MPI
QMC_OMP
QMC_BITS
QMC_COMPLEX
QMC_SK_RECURSIVE
CXX
They are used with $ENV{}

I specified CMAKE_C_COMPILER as Clang but CMake thought it is XL because I have CXX specified in my .bashrc.

CXX is very convenient but can cause hidden problems.
They can be made either cached variables or just removed.

Any thoughts?

**Comment by prckent** [comment]
@naromero77 What are the SPACK requirements for environment variable handling? Do they consider environment variables a good thing?

Our current trajectory has been to add configure options (cmake command line etc.) for everything. Our environment variable handling is certainly inconsistent and not comprehensive. I favor stripping the handling as per @ye-luo since the side effects can be confusing. This would also require updates to the manual and possibly some build scripts.

**Comment by naromero77** [comment]
@prckent OK, I talked to Ye and I understand the question. In the context of the Spack, what we have in their now for passing the compiler to cmake is a bit of hack. The preference is to avoid environment variables and pass the options in through the CMake command line. 

I am fine with getting rid of those environment variables. Not even sure what some of those mean (e.g. QMC_BITS?). Also, it would be wonderful if the CMake build log contained clearly which M^N combinations of options were turned on in a localized place in plain English. A while ago, the only way to verify that certain options were selected where to dig around in a header file. Not very user friendly.
